src/System.Reflection.Metadata/src/System/Reflection/Metadata/MetadataReader.netstandard2.0.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Reflection.Metadata
+{
+    public sealed partial class MetadataReader
+    {
+        internal AssemblyName GetAssemblyName(StringHandle nameHandle, Version version, StringHandle cultureHandle, BlobHandle publicKeyHandle, AssemblyHashAlgorithm assemblyHashAlgorithm, AssemblyFlags flags)
+        {
+            string name = (!nameHandle.IsNil) ? GetString(nameHandle) : null;
+            string cultureName = (!cultureHandle.IsNil) ? GetString(cultureHandle) : null;
+            var hashAlgorithm = (Configuration.Assemblies.AssemblyHashAlgorithm)assemblyHashAlgorithm;
+            byte[] publicKeyOrToken = !publicKeyHandle.IsNil ? GetBlobBytes(publicKeyHandle) : null;
+
+            var assemblyName = new AssemblyName(name)
+            {
+                Version = version,
+                CultureName = cultureName,
+                HashAlgorithm = hashAlgorithm
+            };
+
+            assemblyName.SetPublicKey(publicKeyOrToken);
```

Comment:
@nguerrera  I changed it to 



            bool hasPublicKey = (flags & AssemblyFlags.PublicKey) != 0;
            if (hasPublicKey)
            {
                assemblyName.SetPublicKey(publicKeyOrToken);
            }
            else
            {
                assemblyName.SetPublicKeyToken(publicKeyOrToken);
            }

in my local branch, (haven't pushed it up yet).
but if would fail my test. am I missing something?